src/scanners/langpack.js

```diff
@@ -0,0 +1,51 @@
+import { extname } from 'path';
+
+import FluentParser from 'parsers/fluent';
+import PropertiesParser from 'parsers/properties';
+import DoctypeParser from 'parsers/doctype';
+import BaseScanner from 'scanners/base';
+
+
+export default class LangpackScanner extends BaseScanner {
+  static get scannerName() {
+    return 'langpack';
+  }
+
+  _getContents() {
+    return Promise.resolve(this.contents);
+  }
+
+  scan() {
+    return this.getContents()
+      .then((data) => {
+        const ext = extname(this.filename);
+        let ParserClass = null;
+
+        if (ext === '.properties') {
+          ParserClass = PropertiesParser;
+        } else if (ext === '.ftl') {
+          ParserClass = FluentParser;
+        } else if (ext === '.dtd') {
+          ParserClass = DoctypeParser;
+        } else {
+          throw new Error('Unsupported file type');
+        }
```

Comment:
Are we sure langpacks cannot contain any file types other than the ones listed?

Reply:
The linter only forwards files that are either `.properties`, `.dtd` or `.ftl` and nothing else right now. Practically a langpack can also contain icons and xml because of overwriting search but we don't validate that for right now.